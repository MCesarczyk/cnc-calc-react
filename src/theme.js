export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}