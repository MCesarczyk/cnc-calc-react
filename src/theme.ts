export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
