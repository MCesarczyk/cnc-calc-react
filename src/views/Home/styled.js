import styled from "styled-components";

export const HomeTitle = styled.h2`
  text-align: center;
`;