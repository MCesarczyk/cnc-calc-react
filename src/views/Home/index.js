import languages from "../../assets/fixtures/languages";
import Navigation from "../../features/navigation/Navigation";
import { MobileNavWrapper } from "../../features/navigation/Navigation/styled";
import { HomeTitle } from "./styled";

const Home = ({ langId }) => (
  <>
    <HomeTitle>
      {languages[langId]?.welcomeTitle}
    </HomeTitle>
    <MobileNavWrapper>
      <Navigation langId={langId} />
    </MobileNavWrapper>
  </>
);

export default Home;