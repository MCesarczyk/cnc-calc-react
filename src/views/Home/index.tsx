import { useContext } from 'react';
import styled from 'styled-components';

import LanguageContext from 'features/language/context';
import languages from 'assets/fixtures/languages';
import { MobileNavWrapper, Navigation } from 'features/navigation/Navigation';

const Home = () => {
  const { langId } = useContext(LanguageContext);

  return (
    <>
      <HomeTitle>{languages[langId]?.welcomeTitle}</HomeTitle>
      <MobileNavWrapper>
        <Navigation />
      </MobileNavWrapper>
    </>
  );
};

export const HomeTitle = styled.h2`
  text-align: center;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: none;
  }
`;

export default Home;
