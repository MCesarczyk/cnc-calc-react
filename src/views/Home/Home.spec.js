import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import "jest-styled-components";
import Home from ".";

test("Home component rendered properly", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme} >
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});