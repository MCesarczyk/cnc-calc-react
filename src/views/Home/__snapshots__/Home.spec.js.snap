// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component rendered properly 1`] = `
Array [
  .c0 {
  text-align: center;
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

@media (max-width:768px) {

}

<h2
    className="c0"
  />,
  .c0 {
  display: none;
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

<div
    className="c0"
  />,
]
`;
